### PR TITLE
Remove-QNode.gradient_fn

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
as title
**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

**Related Shortcut Stories:**
[sc-77510]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -20,16 +20,12 @@
...
-      - CX (controlled-X)
-      - Multi-controlled X gates
-      - Grover operators
-  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
+
+  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:
+
+  * Two density matrix contraction methods using `einsum` and `tensordot`
+
+  * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators
```
Comment (JerryChen97) [inline doc/releases/changelog-dev.md:28]:
```suggestion
  - A new API for mixed-state operations
  - An `apply_operation` helper function featuring:
    - Two density matrix contraction methods using `einsum` and `tensordot`
    - Optimized handling of special cases including:
      - Diagonal operators
      - Identity operators 
      - CX (controlled-X)
      - Multi-controlled X gates
      - Grover operators
  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
```

**Comment by JerryChen97** [comment]
@lillian542 @andrijapau

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -39,6 +39,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* ``QNode.gradient_fn`` has been removed. Please use ``QNode.diff_method`` instead. ``QNode.get_gradient_fn`` can also be used to
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:42]:
```suggestion
* `QNode.gradient_fn` has been removed. Please use `QNode.diff_method` instead. `QNode.get_gradient_fn` can also be used to
```

**Review by mudit2812 — APPROVED** [review]

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -50,4 +54,5 @@
 This release contains contributions from (in alphabetical order):
 
 Astral Cai,
+Yushao Chen
```
Comment (andrijapau) [inline doc/releases/changelog-dev.md:57]:
```suggestion
Yushao Chen,
```

File: tests/interfaces/test_autograd_qnode.py [hunk 4]
```diff
@@ -533,33 +533,30 @@ def test_update_diff_method(self, mocker):
...
             qml.RY(a, wires=0)
             qml.RX(b, wires=1)
             qml.CNOT(wires=[0, 1])
             return qml.expval(qml.PauliY(1))
 
-        with pytest.warns(
-            qml.PennyLaneDeprecationWarning, match=r"QNode.gradient_fn is deprecated"
-        ):
-            assert cost_fn.gradient_fn == "backprop"  # gets restored to default
+        # a = cost_fn.get_gradient_fn(dev, "autograd")[0]
```
Comment (andrijapau) [inline tests/interfaces/test_autograd_qnode.py:545]:
```suggestion
```

Accidentally left a comment here.

**Review by andrijapau — APPROVED** [review]
Nice!

**Comment by JerryChen97** [comment]
closed due to the bundled PR already in

**Comment by Alex-Preciado** [comment]
Closed in favour of https://github.com/PennyLaneAI/pennylane/pull/6535

**Pull request closed without merging**